common/src/main/java/io/apicurio/registry/rest/ArtifactsResource.java

```diff
@@ -27,6 +27,19 @@
  */
 @Path("/artifacts")
 public interface ArtifactsResource {
+  /**
+   * Test if artifact is compatible with previous versions.
+   *
+   * If the test content violates one of the rules configured for the artifact you'll receive a HTTP error `400`.
+   * If all is OK, HTTP 200 code is returned.
+   */
+  @Path("/{artifactId}/test")
+  @PUT
+  @Consumes("application/json")
+  void testCompatibility(@PathParam("artifactId") String artifactId,
```

Comment:
This should be something like "testRules" because it could be more than just compatibility.  It will apply all rules, which now consists of validation and compatibility, but in the future could be even more.  Don't make this change here though, I'll update the OpenAPI definition and re-generate the jax-rs layer.  When I do that I'll rename this and fix the impl.

Reply:
OK, makes sense.
What about then the name of the goal? Currently it's "compatibility".
Rename that as well, the way you see it fit.

Reply:
@EricWittmann perhaps a simple "test" or "test-rules" ?